app/workers/delete_payment_setting_hierarchy_worker.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class DeletePaymentSettingHierarchyWorker < DeleteObjectHierarchyWorker
+  alias payment_setting object
+
+  def delete_associations
+    destroy_users_of_provider(payment_setting.account) if called_from_hierarchy?
+
+    super
```

Comment:
Does this mean that we will enqueue the deletion of each buyer account twice when a provider account is deleted?

Reply:
So far yes but I intend to change that by slightly changing the `DeleteAccountHierarchyWorker` :smile: But I am focused in making it work first :disappointed: